arithmetization/src/main/java/net/consensys/linea/sequencer/txselection/selectors/TraceLineLimitTransactionSelector.java

```diff
@@ -151,6 +171,21 @@ public BlockAwareOperationTracer getOperationTracer() {
     return zkTracer;
   }
 
+  private void rememberOverLineCountLimitTransaction(final Transaction transaction) {
+    if (overLineCountLimitCache.size() >= overLimitCacheSize) {
+      final var it = overLineCountLimitCache.iterator();
+      if (it.hasNext()) {
```

Comment:
non-blocking, but IMO it would be more robust to make this a while loop on the off chance that the cache is more than one entry over the limit for any reason.

Reply:
done